src/profiler.cpp

```diff
@@ -1677,9 +1709,28 @@ void Profiler::dumpOtlp(Writer& out, Arguments& args) {
         CallTrace* trace = cts->acquireTrace();
         if (trace == NULL || excludeTrace(&fn, trace) || cts->samples == 0) continue;
 
+        const char* trace_id = nullptr;
+        const char* span_id = nullptr;
+
+        for (int j = 0; j < trace->num_frames; j++) {
+            if (trace->frames[j].bci == BCI_TRACE_ID) {
+                trace_id = (const char*)trace->frames[j].method_id;
+            }
+            if (trace->frames[j].bci == BCI_SPAN_ID) {
+                span_id = (const char*)trace->frames[j].method_id;
+            }
+        }
+
         protobuf_mark_t sample_mark = otlp_buffer.startMessage(Profile::sample, 1);
         otlp_buffer.field(Sample::locations_start_index, frames_seen);
         otlp_buffer.field(Sample::locations_length, trace->num_frames);
+
+        if (trace_id && span_id) {
+            std::string link_key = std::string(trace_id) + ":" + std::string(span_id);
+            size_t link_idx = links.indexOf(link_key);
+            otlp_buffer.field(Sample::link_index, link_idx + 1);
```

Comment:
The first idx you get out of `Index` is always going to be 1, check the implementation

Reply:
```suggestion
            otlp_buffer.field(Sample::link_index, link_idx);
```